testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/SuiteContext.java

```diff
@@ -58,7 +60,9 @@
      * True if the testsuite is running in the adapter backward compatibility testing mode,
      * i.e. if the tests are running against newer auth server
      */
-    private static final boolean adapterCompatTesting = Boolean.parseBoolean(System.getProperty("testsuite.adapter.compat.testing"));
+    private static final boolean adapterCompatTesting = parseBoolean(System.getProperty("testsuite.adapter.compat.testing"));
+
+    private static final boolean browserStrictCookies = parseBoolean(System.getProperty("browser.strict.cookies"));
```

Comment:
Not sure this has any other simple solution but tests using this boolean will start failing once chrome will block 3rd party cookies by default. Do you have any idea, how to avoid this? Maybe some simple app with iframe within testsuite-providers which will check this? WDYT?

Reply:
```suggestion
    private static final boolean browserStrictCookies = parseBoolean(System.getProperty("browser.strict.cookies")) && !"localhost".equals(AUTH_SERVER_HOST);
```

Reply:
We won't be testing this with Chrome for now. We'll be using FF in controlled CI environment where we're using a stable version, not relying on auto updates.

Reply:
These tests are running in server pipeline with chrome. But I just realized, we can rely on the fact that server pipeline is using `localhost` by default.

Reply:
Yes, with localhost cookies are not blocked.

Reply:
Anyway, I think this part `&& !"localhost".equals(AUTH_SERVER_HOST);` makes sense because if you run tests without `auth.server.host` or/and (not sure) `auth.server.host2`, tests will fail, because 3rd party are not blocked. Or maybe we can set some default values for hosts when `browser.strict.cookies` == true. Maybe using nip.io/xip.io WDYT?

Reply:
I wouldn't add the suggested comparison to localhost here. The `browserStrictCookies` property just tells us the browser treats cookies in some way and doesn't tell us when they'll be blocked. It's up to the test to know it.

Also, when the tests fail without setting proper hosts for server, it's actually correct behavior IMHO. It points out a miscofigured test.

But I agree maybe we could add a special Maven profile to set hosts to some precofigured values. But I'm not sure about it – if we want to hardcode a 3rd party services directly into POM. IMHO that belongs to the test pipeline.

Reply:
Well, I think it should fail sooner than in the test if one fails to set hosts properly. However, it would be harder to implement it as we don't know whether we should enforce `auth.server.host` + `app.server.host` or `auth.server.host` + `auth.server.host2`. Unless we unify those two and use only `auth/app.server.host`. Do you have any idea how to solve it? If no we can probably leave it as it is.